src/wp-includes/ms-site.php

```diff
@@ -498,7 +498,16 @@ function wp_normalize_site_data( $data ) {
 	// Sanitize domain if passed.
 	if ( array_key_exists( 'domain', $data ) ) {
 		$data['domain'] = trim( $data['domain'] );
+		$has_ports = strstr( $data['domain'], ':' );
+
+		$data['domain'] = preg_replace( '|:\d+$|', '', $data['domain'] ); // Remove ports.
 		$data['domain'] = preg_replace( '/\s+/', '', sanitize_user( $data['domain'], true ) );
+		/** This filter is documented in wp-includes/ms-settings.php */
+		$allowed_ports = apply_filters( 'allowed_multisite_ports', array( ':80', ':443' ) );
+		if ( ( false !== $has_ports && in_array( $has_ports, $allowed_ports ) ) ) {
```

Comment:
```suggestion
		if ( ( false !== $has_ports && is_array( $allowed_ports ) && in_array( $has_ports, $allowed_ports, true ) ) ) {
```
- We call `in_array()` without confirming that `$allowed_ports` is an array after filtering. We should add this guard.
- Adding `true` to `in_array()` will make the check strict. `in_array()` should perform a strict check where possible.